### PR TITLE
[CORE] Update Substrait to 0.24.0

### PR DESCRIPTION
**Pull request by ted-jenks**
## What changes were proposed in this pull request?
Update Substrait to 0.24.0

I understand a lot of [this](https://github.com/oap-project/gluten/pull/515) is currently dead code, but would be nice to move things in the right direction.

## How was this patch tested?
Existing tests.

Local Package Test:
```bash
mvn clean package -Pbackends-velox -Pspark-3.4 -Prss -DskipTests
```

### CODE REVIEW

File: substrait/substrait-spark/src/main/scala/io/substrait/spark/logical/ToSubstraitRel.scala [hunk 1]
```diff
@@ -327,6 +319,27 @@ class ToSubstraitRel extends AbstractLogicalPlanVisitor with Logging {
         ))
       .build()
   }
+
+  def tree(p: LogicalPlan): String = {
+    TreePrinter.tree(visit(p))
+  }
+
+  def toProtoSubstrait(p: LogicalPlan): Array[Byte] = {
```
Comment (ted-jenks) [inline substrait/substrait-spark/src/main/scala/io/substrait/spark/logical/ToSubstraitRel.scala:327]: This method allows for creation of a protobuf plan representation from the spark plan.

**Comment by ted-jenks** [comment]
@PHILO-HE please may I have a review?

**Comment by PHILO-HE** [comment]
The red CI may have been fixed in main branch. Please do a rebase. Thanks!

**Comment by PHILO-HE** [comment]
@baibaichen

File: pom.xml [hunk 2]
```diff
@@ -68,7 +68,7 @@
     <!-- Package name to use when relocating shaded classes -->
     <gluten.shade.packageName>io.glutenproject.shaded</gluten.shade.packageName>
 
-    <substrait.version>0.5.0</substrait.version>
+    <substrait.version>0.22.0</substrait.version>
```
Comment (zhouyuan) [inline pom.xml:71]: 0.22.0 seems to be one old tag, curious why use this one?

**Review by zhouyuan — COMMENTED** [review]
Thanks for improving on this! The substrait-spark is not really used in gluten, but it's ok to me if all CI is green. 
CC @zzcclp

File: pom.xml (hunk 2 above)
Comment (ted-jenks) [reply pom.xml:71]: Ah I didn't realize there was another release. I will update.

**Comment by ted-jenks** [comment]
I merged master and it broke 🥴

**Comment by PHILO-HE** [comment]
> I merged master and it broke 🥴

Don't worry. It's not relevant to your pr. The log says the disk has no space. Let me handle it.

**Review by zhouyuan — APPROVED** [review]
👍

**Pull request merged**